### ISSUE TITLE
404 error when accessing chat room from public side

### ISSUE THREAD

**Issue by IonicaBizau** [gitterHQ/gitter#1340]
It would be really nice to show a friendly error if the room exists, but you have to be logged in to access it.

This applies when the room access is limited to the organization members.

**Comment by MadLittleMods** [comment]
As a note, we show this friendly message for rooms that may require private repo access but this is when you are signed in already.

![](http://i.imgur.com/5Eu8qCJ.png)

**Comment by IonicaBizau** [comment]
@MadLittleMods Yes, it would be nice to have it even when you are *not* signed in.

**Comment by cbj4074** [comment]
The Gitter developers have explained that the behavior is simply in keeping with GitHub's (ill-advised, in my opinion) decision to return a 404, instead of a 403, when the user-agent attempts to access a resource that does exists, but the user lacks access (hasn't logged in, or simply lacks the required permissions).

One can only assume that this is a "security feature" and/or a "privacy feature". In GitHub's defense, revealing that a resource even exists when the user-agent lacks access does seem like a privacy breach.

But the current behavior is annoying as hell.

When the user is logged-in but doesn't have access, show a 404 (in keeping with above-noted privacy concerns).

When the user is not logged-in, show a 403, with a message: "This resource may or may not exist. Log in to gain access."

Wouldn't that be favorable as compared to the current behavior?

Am I overlooking something?

**Comment by brusznicki** [comment]
Agree. This is exceptionally annoying when viewing a gh resource linked from email or slack On a mobile device.
